docs/UPGRADE.md

```diff
@@ -0,0 +1,207 @@
+# Upgrade
+
+## Upgrade guide `3.x.x` to `4.x.x`
+
+This upgrade is mainly caused by upgrading [Kentico Kontent Javascript Delivery SDK](https://github.com/Kentico/kontent-delivery-sdk-js) and adding new features.
+
+### New Featuers
+
+* Language fall backs
+* enableLogging configuration
+
+### Breaking changes
+
+* Query names KenticoCloud -> Kontent
+* prefered_language
+* value in fields
+
+### Delivery config
+
+When configuring the Kentico Kontent Source plugin one of the properties to set is `deliveryClientConfig`. It is respecting the [`IDeliveryClientConfig`](https://github.com/Kentico/kontent-delivery-sdk-js/blob/master/UPGRADE.md#ideliveryclientconfig) interface from Kentico Kontent Delivery SDK.
+
+#### Example
+
+```javascript
+module.exports = {
+  ...
+  plugins: [
+    ...
+    {
+      resolve: `gatsby-source-kentico-cloud`,
+      options: {
+        deliveryClientConfig: { // Configuration object
+          projectId: `XXX`,
+          typeResolvers: [],
+          previewApiKey: `YYY`,
+          secureApiKey: `ZZZ`,
+          globalQueryConfig:  {
+            ...
+            usePreviewMode: true, // uses preview mode
+            useSecuredMode: false, // disabled secured mode
+            ...
+          },
+        },
+        languageCodenames: [ // example configuration
+          `en-US`, // default language
+          `es-ES`,
+        ]
+      }
+    }
+    ...
+  ]
+  ...
+}
+```
+
+### Element value property unification
+
+All of the elements types [has element property called `value`](https://github.com/Kentico/kentico-cloud-js/blob/master/packages/delivery/UPGRADE.md#removal-of-type-specific-element-properties) not an element specific property.
+
+Custom element is now supported including [custom element models definition](https://github.com/Kentico/kentico-cloud-js/blob/master/packages/delivery/DOCS.md#using-custom-models-for-custom-elements).
```

Comment:
link does not work vs kontent rename

Reply:
Should be: https://github.com/Kentico/kontent-delivery-sdk-js/blob/master/DOCS.md#using-custom-models-for-custom-elements